theme/static/js/hs_resource_table.js

```diff
@@ -533,6 +579,71 @@ function updateLabelsList() {
     }
 }
 
+// Gets new data when filters change
+async function dataRefresh(target){
+    
+    // Check the url for query parameters
+    let url = new URL(window.location.href);
+    let filter_val = target.value.toLowerCase();
+    let existing_filters = url.searchParams.getAll('f');
+    if (existing_filters){
+        let index = existing_filters.indexOf(filter_val);
+        if(index > -1){
+            if (!target.checked){
+                existing_filters.splice(index, 1);
+                url.searchParams.delete('f');
+                existing_filters.forEach(filter => url.searchParams.append('f', filter));
+            }
+        }else{
+            if (target.checked){
+                url.searchParams.append('f', filter_val);
+            }
+        }
+    }
+
+    window.history.pushState({}, '', url);
+
+    if(target.checked && !filtersLoaded.includes(filter_val)){
+        // fetch the filter that is new
+        await ajaxFetchResources(url, filter_val);
+    }else{
+        updateTable();
+    }
+}
+
+function ajaxFetchResources(url, filter_val){
+    $("#filter-panel .panel-title").append(spinner);
+    $("#filter").addClass("no-interact")
+    $('#item-selectors tbody').hide();
+    $('#item-selectors thead').hide();
+
+    $.ajax({
+        type: 'GET',
+        url: url,
+        data: {
+            new_filter: filter_val
+        }
+    })
+    .done(function(data){
+        resourceTable.destroy();
+        $('#item-selectors tbody').append(data.trows);
+        resourceTable = initTable('#item-selectors');
```

Comment:
@Maurier this `initTable()` causes the column widths to get garbled ("Type" col gets really wide). I thought maybe `resourceTable.draw()` would fix that, but doesn't seem to. Do you have a simple suggestion? Thanks

Reply:
disregard, I found
https://datatables.net/reference/option/autoWidth